docs/cel2/operators/migrate-node.md

```diff
@@ -112,7 +112,39 @@ Alternatively, you can run the check-db script on its own by checking out the la
 
 ## Migrating a Celo archive node
 
-Node operators who were running archive nodes before the migration and wish to maintain execution
-and state access functionality for pre-hardfork blocks will need to continue to run their L1 node
-and configure their L2 node to proxy pre-hardfork execution and state access requests to the L1 node
-by setting the `OP_GETH__HISTORICAL_RPC` in `.env` to the RPC address of their L1 node.
+We recommend only running the migration on full node data, and not on archive node data. If you only have L1 archive nodes, we recommend syncing a L1 full node for the Mainnet migration.
+
+The L2 execution client cannot use pre-hardfork state data, so migrating an archive datadir will just copy large amounts of data unnecessarily. The migration script will also run slowly and consume large amounts of memory when run on archive data. For these reasons, we recommend only running the migration script on a full node L1 datadir even if you plan to run an L2 archive node.
+
+To run an L2 archive node, you should migrate from an L1 full node datadir but still start the L2 execution client in archive mode. This will allow the node to accept rpc requests that require an archive node. Then, you should set `OP_GETH__HISTORICAL_RPC` in `.env` to the RPC address of a running legacy L1 archive node. The L2 node will proxy requests for pre-hardfork state and execution to the legacy L1 archive node instead of looking for the archive state in its own database. The L2 node will still run as a standard archive node storing post-hardfork data.
+
+Here are step-by-step instructions for using [celo-l2-node-docker-compose](https://github.com/celo-org/celo-l2-node-docker-compose) to run an archive L2 node:
```

Comment:
This is very good to have. Two general remarks:
- I'd stick to the imperative throughout the instructions for brevity (eg: shorten "At this point, you should be able to validate..." to "Validate...")
- Make the instructions copy-pastable with code blocks instead of inline (eg: https://docs.celo.org/network/node/run-mainnet)

Reply:
Updated with the imperatives. The rest of the docs use inline commands throughout so I just did the same for consistency? Would you like me to update the rest of the docs to use code blocks? Should this be in a new PR?